src/AsyncTestCase.php

```diff
@@ -60,6 +60,16 @@ abstract class AsyncTestCase extends PHPUnitTestCase
                 $invoked = true;
                 $exception = $error;
                 $returnValue = $value;
+
+                if ($exception) {
+                    Loop::unreference(Loop::delay(1000, function () use ($exception) {
+                        Loop::stop();
+                        $this->fail(
+                            'An exception was thrown from the test method or promise returned from test method failed),'
+                            . 'but the event loop continued to run; ' . $exception
+                        );
+                    }));
+                }
```

Comment:
Shouldn't this immediately fail?

Reply:
The delay is to differentiate between a test that simply fails (or has an expected exception) vs. a test that hangs after throwing an exception because the loop continues to run.

Reply:
But then a delay(1) should be enough - the loop won't be continued anyway under normal circumstances once it has thrown?

Reply:
The loop didn't throw, rather the test method threw or the promise returned failed.

1ms is probably too short, but 1000ms is also probably too long. 100ms would probably be more reasonable.

Reply:
A defer watcher is probably enough?

Reply:
My reasoning for the delay rather than a defer is that I assumed some libraries such as parallel may have tests that fail the returned promise and then need some time to finish executing, such as shutting down a child process or thread.

After trying `amphp/cluster` with a defer watcher, I found that `WatcherTest::testDoubleStart` was failing because stopping the cluster was taking more than a couple seconds. This was the only test I found failing among several libraries I tried (amp, parallel, http-server, postgres).

I pushed another commit that uses the timeout if set, otherwise uses 1ms.

I'm not sure if we can consider this backward-compatible, as the cluster test passes with 1.2.x. 

This issue may be better handled with documentation on hanging loops. 🤷‍♂

Reply:
We could also release it as 2.0.